[smartmeter] NPE while trying to find channel

```
2021-01-31 15:45:56.276 [ERROR] [ding.smartmeter.internal.MeterDevice] - Meter listener failed
java.lang.NullPointerException: null
        at org.openhab.binding.smartmeter.internal.SmartMeterChannelTypeProvider.getChannelTypeIdForObis(SmartMeterChannelTypeProvider.java:110) ~[bundleFile:?]
        at org.openhab.binding.smartmeter.internal.SmartMeterHandler$1.valueChanged(SmartMeterHandler.java:168) ~[bundleFile:?]
        at org.openhab.binding.smartmeter.internal.MeterDevice.lambda$9(MeterDevice.java:220) [bundleFile:?]
        at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:807) [?:?]
        at org.openhab.binding.smartmeter.internal.MeterDevice.addObisCache(MeterDevice.java:218) [bundleFile:?]
        at org.openhab.binding.smartmeter.internal.iec62056.Iec62056_21MeterReader.populateValueCache(Iec62056_21MeterReader.java:58) [bundleFile:?]
        at org.openhab.binding.smartmeter.internal.iec62056.Iec62056_21MeterReader.populateValueCache(Iec62056_21MeterReader.java:1) [bundleFile:?]
        at org.openhab.binding.smartmeter.internal.MeterDevice.lambda$6(MeterDevice.java:184) [bundleFile:?]
        at io.reactivex.internal.subscribers.LambdaSubscriber.onNext(LambdaSubscriber.java:65) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.onNext(FlowableSubscribeOn.java:97) [bundleFile:?]
        at io.reactivex.subscribers.SerializedSubscriber.onNext(SerializedSubscriber.java:100) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableRepeatWhen$WhenSourceSubscriber.onNext(FlowableRepeatWhen.java:159) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableRefCount$RefCountSubscriber.onNext(FlowableRefCount.java:199) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowablePublish$PublishSubscriber.dispatch(FlowablePublish.java:547) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowablePublish$PublishSubscriber.onNext(FlowablePublish.java:231) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableDoOnEach$DoOnEachSubscriber.onNext(FlowableDoOnEach.java:92) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle$SubscriptionLambdaSubscriber.onNext(FlowableDoOnLifecycle.java:79) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableDoOnEach$DoOnEachSubscriber.onNext(FlowableDoOnEach.java:92) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle$SubscriptionLambdaSubscriber.onNext(FlowableDoOnLifecycle.java:79) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onNext(FlowableTimeoutTimed.java:101) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableCreate$NoOverflowBaseAsyncEmitter.onNext(FlowableCreate.java:403) [bundleFile:?]
        at org.openhab.binding.smartmeter.internal.iec62056.Iec62056_21SerialConnector$1.newDataMessage(Iec62056_21SerialConnector.java:103) [bundleFile:?]
        at org.openmuc.j62056.Iec21Port$ModeDReceiver.run(Iec21Port.java:71) [bundleFile:?]
```

--- a/bundles/org.openhab.binding.smartmeter/src/main/java/org/openhab/binding/smartmeter/internal/SmartMeterChannelTypeProvider.java
+++ b/bundles/org.openhab.binding.smartmeter/src/main/java/org/openhab/binding/smartmeter/internal/SmartMeterChannelTypeProvider.java
@@ -107,6 +107,7 @@ public class SmartMeterChannelTypeProvider implements ChannelTypeProvider, Meter
      * @return The {@link ChannelTypeUID} or null.
      */
     public ChannelTypeUID getChannelTypeIdForObis(String obis) {
-        return obisChannelMap.get(obis).getUID();
+        ChannelType channeltype = obisChannelMap.get(obis);
+        return channeltype != null ? channeltype.getUID() : null;
     }
 }

